messaging/relayer/internal/messages/raw.go

```diff
@@ -0,0 +1,154 @@
+package messages
+
+/*
+ * Copyright 2019 ConsenSys Software Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+)
+
+// RawMessage represents an undecoded message.
+type RawMessage struct {
+	id        cid.Cid
+	timestamp int64
+	version   string
+	msgType   int64
+	payload   []byte
+}
+
+// rawMessageJson is used for serialization.
+type rawMessageJson struct {
+	ID        string `json:"id"`
+	Timestamp int64  `json:"timestamp"`
+	Version   string `json:"version"`
+	MsgType   int64  `json:"msg_type"`
+	Payload   string `json:"payload"`
+}
+
+// NewRawMessage creates a raw message.
+func NewRawMessage(version string, msgType int64, payload []byte) Message {
+	now := time.Now().Unix()
+	// Generate unique id
+	data, _ := json.Marshal(rawMessageJson{
```

Comment:
Aha, yes, maybe adding a bit of check is better. The reason I didn't check error is that json.Marshal does not fail as long as the data structure is not cyclic. Same for the CID generation, it shouldn't fail based on the way I use it. I could maybe panic if error occurs or what do you think? (I was just trying to avoid writing lines that are can not be tested)

Reply:
Right, panic makes sense.